trestle/tasks/csv_to_oscal_cd.py

```diff
@@ -0,0 +1,292 @@
+                    uuid=str(uuid.uuid4()),
+                    source=source,
+                    description=description,
+                    implemented_requirements=[],
+                )
+                control_implementations[control_id] = control_implementation
+                defined_component.control_implementations.append(control_implementation)
+        # implemented requirements
+        for row in self.csv_helper.row_generator():
+            control_id = self.csv_helper.get_value(row, 'Control_Mappings')
+            control_implementation = control_implementations[control_id]
+            implemented_requirement = ImplementedRequirement(
+                uuid=str(uuid.uuid4()),
+                control_id=control_id,
+                description=control_id,
+                props=[],
+            )
+            implemented_requirements[control_id] = implemented_requirement
+            control_implementation.implemented_requirements.append(implemented_requirement)
+        # rules
+        index = 0
+        for row in self.csv_helper.row_generator():
+            control_id = self.csv_helper.get_value(row, 'Control_Mappings')
+            implemented_requirement = implemented_requirements[control_id]
+            ns = self._ns
+            remarks = f'rule_set_{str(index).zfill(3)}'
+            # Rule_Id
+            name = 'Rule_Id'
+            value = self.csv_helper.get_value(row, name)
+            prop = Property(
+                name=name,
+                value=value,
+                ns=ns,
+                remarks=remarks,
+            )
+            # Rule_Description
+            name = 'Rule_Description'
+            value = self.csv_helper.get_value(row, name)
+            prop = Property(
+                name=name,
+                value=value,
+                ns=ns,
+                remarks=remarks,
+            )
+            implemented_requirement.props.append(prop)
+            index += 1
+
+    def _get_catalog_title(self) -> str:
+        """Get catalog title."""
+        value = 'TBD'
+        return value
+
+    def _get_org_name(self) -> str:
+        """Get org name."""
+        value = 'TBD'
+        return value
+
+    def _get_org_remarks(self) -> str:
+        """Get org remarks."""
+        value = 'TBD'
+        return value
+
+    def _get_components(self) -> List[DefinedComponent]:
+        """Get components."""
+        value = self._components
+        return value
+
+    def _build_roles(self) -> List[Role]:
+        """Build roles."""
+        value = [
+            Role(id='prepared-by', title='Indicates the organization that created this content.'),
+            Role(id='prepared-for', title='Indicates the organization for which this content was created..'),
+            Role(
+                id='content-approver',
+                title='Indicates the organization responsible for all content represented in the "document".'
+            ),
```

Comment:
Are these hardcoded for every cd?

Reply:
Fixed, by removal as prescribed by Vikas.